include/picongpu/fields/MaxwellSolver/Yee/Yee.kernel

```diff
@@ -47,6 +48,11 @@ namespace picongpu
                 class UpdateEFunctor
                 {
                 public:
+                    //! Stencil requirements for lower margins
+                    using LowerMargin = typename traits::GetLowerMargin<T_CurlB>::type;
+                    //! Stencil requirements for upper margins
+                    using UpperMargin = typename traits::GetUpperMargin<T_CurlB>::type;
+
```

Comment:
I think it is worth to add a base class for stencils that takes `Curl` as template, and defines those margins. Besides removing the duplication of those, it would allow clarifying stencil requirements in the docs for the update field kernel.

Reply:
Each class functor has an implementation and if the stencil would use the curl plus something else the margin would be `max(curl, otherStancilOperator)`. You are right currently all implementations looks equal but only because the functor implementation is using one stancil. I do not think it helps to implement somewhere a class which collects upper and lower margin. The current definition of the margins via in-class definition is only one way, I could also specialize the trade for margins, in this case lower and upper margin would be independent too.

Reply:
Not sure if I agree. With how this part now functions, it can only solve equations in form Field += Curl(another field). So curl is the only parameter here kinda. If we wanted to support another structure of equations to solve, that would require modifications anyways.

My motivation to have a base class is as follows:

- More importanly, having a base class with a telling name allows using this name in the doxygen comments. And it makes it searchable in the code base.
- Less importantly, to remove duplication

Reply:
That said, chances are this is not most actively changed part of the code, after the ongoing modernization is finished. So i don't insist on that.

Reply:
Where would you place this unified class and how would you name it. Yee functors and PML functors would use this class, for me, it sounds like we mix something. 
Could you maybe open a PR against my PR, maybe I understand something wrong and it gets clear when I see your intention in code.

Reply:
E.g. `StelcilFunctor`. Note that now (as in your PR currently) it's not easily possible to find implementations of this functor by looking on the update field kernel. None of the functors have "stencil" at their name, nor their main doxygen string (only inside the class).

Reply:
Yes, perhaps I did not express my idea well. Will do, or discuss in mattermost for more simple communication